### PR TITLE
Harden container/shim/src/index.js

### PR DESCRIPTION
**Pull request by juliangruber**
I skipped the express specific parts for this review because of https://github.com/filecoin-saturn/L1-node/issues/33. Hope you find this useful!

### CODE REVIEW

File: container/shim/src/index.js [hunk 1]
```diff
@@ -46,15 +48,17 @@ if (cluster.isPrimary) {
   process.on('SIGQUIT', shutdownCluster)
   process.on('SIGINT', shutdownCluster)
 
-  setTimeout(async function () {
-    await register(true).catch(err => {
-      debug(`Failed to register ${err.name} ${err.message}`)
-      process.exit(1)
-    })
+  await timers.setTimeout(500)
```
Comment (juliangruber) [inline container/shim/src/index.js:51]: safer to use the promise version

File: container/shim/src/index.js [hunk 2]
```diff
@@ -46,15 +48,17 @@ if (cluster.isPrimary) {
...
+
+  try {
+    await register(true)
+  } catch (err) {
+    debug(`Failed to register ${err.name} ${err.stack}`)
+    process.exit(1)
+  }
```
Comment (juliangruber) [inline container/shim/src/index.js:58]: To me it's easier to basic the standard language features

File: container/shim/src/index.js [hunk 3]
```diff
@@ -91,9 +95,10 @@ if (cluster.isPrimary) {
     })
 
     if (req.headers.range) {
-      let [start, end] = req.headers.range.split('=')[1].split('-')
-      start = parseInt(start, 10)
-      end = parseInt(end, 10)
+      const range = req.headers.range.split('=')?[1]?.split('-')
```
Comment (juliangruber) [inline container/shim/src/index.js:98]: an invalid range header could have crashed the process I think

File: container/shim/src/index.js [hunk 4]
```diff
@@ -91,9 +95,10 @@ if (cluster.isPrimary) {
...
+      assert(range, 'TODO')
+      const start = Number(range[0])
+      const end = Number(range[0])
```
Comment (juliangruber) [inline container/shim/src/index.js:101]: `Number` is safer than `parseInt` because you can't forget the radix

File: container/shim/src/index.js [hunk 5]
```diff
@@ -120,6 +125,8 @@ if (cluster.isPrimary) {
       controller.abort()
     }, GATEWAY_TIMEOUT)
 
+    // Use node-fetch or undici instead, the `"error"`
+    // event makes this difficult to consume
```
Comment (juliangruber) [inline container/shim/src/index.js:129]: I didn't know which way you wanted to go, but I think it is important to use a promise native request library.

File: container/shim/src/index.js [hunk 6]
```diff
@@ -91,9 +95,10 @@ if (cluster.isPrimary) {
     })
 
     if (req.headers.range) {
-      let [start, end] = req.headers.range.split('=')[1].split('-')
-      start = parseInt(start, 10)
-      end = parseInt(end, 10)
+      const range = req.headers.range.split('=')?[1]?.split('-')
+      assert(range, 'TODO')
```
Comment (DiegoRBaquero) [inline container/shim/src/index.js:99]: Do we want to fail on invalid range? Or return empty bytes?

Comment (juliangruber) [reply container/shim/src/index.js:99]: I think a 4xx response is appropriate, so the client can know they need to adjust their request

Comment (DiegoRBaquero) [reply container/shim/src/index.js:99]: Then we need either try catch or do a truthy check returning 400

Comment (DiegoRBaquero) [inline container/shim/src/index.js:99]:
```suggestion
      if(!range) {
        return res.status(400).send('Invalid range')
      }
```

Comment (juliangruber) [reply container/shim/src/index.js:99]: This is where the error handling comes in. I actually meant to use https://www.npmjs.com/package/http-assert:

```suggestion
      httpAssert(range, 400, 'invalid range')
```

If the async route handler now forwards this to `next`, then the express final middleware can take care of responding with the appropriate error per request

Comment (DiegoRBaquero) [reply container/shim/src/index.js:99]: Can you add the update it with it?

**Pull request closed without merging**